### PR TITLE
Beef up docs about integrating Cargo with other stuff

### PR DESCRIPTION
**Pull request by matklad**
Add info about `--message-format=json`, `--dep-info` and custom cargo subcommands (mostly copy-pasted from the wiki :) ).

cc @carols10cents 

r? @alexcrichton

### CODE REVIEW

File: src/doc/header.html [hunk 1]
```diff
@@ -40,7 +40,7 @@
                 <li><a href='pkgid-spec.html'>Package ID specs</a></li>
                 <li><a href='environment-variables.html'>Environment Variables</a></li>
                 <li><a href='source-replacement.html'>Source Replacement</a></li>
-                <li><a href='machine-readable-output.html'>Machine readable output</a></li>
+                <li><a href='external-tools.html'>External Tools</a></li>
```
Comment (matklad) [inline src/doc/header.html]: Note to self: this should be updated on the `crates.io` side as well.

File: src/doc/external-tools.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+Information about dependencies in the Makefile-compatible format is stored in
+the `.d` files alongside the artifacts.
+
+
+# Custom subcommands.
+
+Cargo is designed to be extensible with new subcommands without having to modify
+Cargo itself. This is achieved by translating a cargo invocation of the form
+cargo `(?<command>[^ ]+)` into an invocation of an external tool
+`cargo-${command} that then needs to be present in one of the user's `$PATH`
```
Comment (alexcrichton) [inline src/doc/external-tools.md]: Missing backtick here I think?

**Comment by alexcrichton** [comment]
Looks great to me! I think the makefile may also need to be updated for doc generation?

Other than that r=me

**Pull request merged**